### DISCUSSION TITLE
Question about Exchange Spec.Name vs. ObjectMeta.Name

### DISCUSSION THREAD

**Discussion by vaikas** [rabbitmq/messaging-topology-operator#100] [General]
Is this field:
https://github.com/rabbitmq/messaging-topology-operator/blob/main/api/v1alpha2/exchange_types.go#L21
And this:
https://github.com/rabbitmq/messaging-topology-operator/blob/main/api/v1alpha2/queue_types.go#L26

Different from {Exchange, Queue}.ObjectMeta.Name?

If not, just curious if it's needed as it seems redundant. If it is, how does it handle collisions if two CRs create an exchange (or queue)with different k8s name but use the same Spec.Name field?

Thanks!

**Comment by ChunyiLyu** [comment]
I know this setup would be confusing to people😄 Perhaps documentation could be improved. The reasons for not taking objectMeta.name as the object name is: 1) `spec.rabbitmqClusterReference` is a `localObjectReference` object, so all objects have to be created in the same namespace as their corresponding Rabbit cluster. 2) Objects in RabbitMQ are namespaced by their vhosts. You can have exchanges, queues, and policies created with the same name but in different vhosts. If we were to take cr.ObjectMeta.Name as the object name, it means that exchanges, queues, and policies need to have unique names despite they are in different vhosts (as stated before, objects of one Rabbit cluster need to be created in the same k8s namespace).

**Reply by gvmw** [reply]
How does https://github.com/knative-sandbox/eventing-rabbitmq/pull/492 affect this discussion?

Even though we have started with `Exchange` first, this is the change that we are driving across all dependent objects (Exchange, Queue & Binding): https://github.com/knative-sandbox/eventing-rabbitmq/pull/492/files#diff-8dfee408e58702a5d050d831118989414786268bc8de253910155a1cf5504d5cR65-R66
